File: WarehouseManagementWeb.Application/Dto/Output/Resource/ResourceListByStatusOutput.cs
using WarehouseManagementWeb.Domain.Entities;

namespace WarehouseManagementWeb.Application.Dto.Output.Resource
{
    /// <summary>
    /// Класс выходной модели ресурсов разделенных по статусам.
    /// </summary>
    public class ResourceListByStatusOutput
    {
        /// <summary>
        /// Список активных ресурсов.
        /// </summary>
        public IEnumerable<ResourceEntity>? ActiveResources { get; set; }

        /// <summary>
        /// Список ресурсов находящихся в архиве.
        /// </summary>
        public IEnumerable<ResourceEntity>? ArchivedResources { get; set; }
    }
}

File: warehouse-management-web-angular/src/app/directory/resource/models/output/resource-list-by-status-output.ts
import { ResourceOutput } from "./resource-output";

/**
 * Класс выходной модели ресурсов разделенных по статусам.
 */
export class ResourceListByStatusOutput {
  /**
   * Список активных ресурсов.
   */
  activeResources: ResourceOutput[] = [];

  /**
   *Список ресурсов находящихся в архиве.
   */
  archivedResources: ResourceOutput[] = [];
}
